python/ray/serve/handle.py

```diff
@@ -344,9 +346,12 @@ async def _fetch_future_result_async(self) -> ReplicaResult:
         if self._replica_result is None:
             # Use `asyncio.wrap_future` so `self._replica_result_future` can be awaited
             # safely from any asyncio loop.
-            self._replica_result = await asyncio.wrap_future(
-                self._replica_result_future
-            )
+            try:
+                self._replica_result = await asyncio.wrap_future(
+                    self._replica_result_future
+                )
+            except asyncio.CancelledError:
+                raise RequestCancelledError from None
```

Comment:
Doesn't this take the request ID as an arg?

Reply:
hmmm yeah, it's optional though. let me see if we can get ahold of the request metadata here.

Reply:
added request ID, ptal!

Reply:
maybe let's make it required so this is consistent